### DISCUSSION TITLE
Zusatzsoftware nur mit zusätzlichem Login sichtbar?

### DISCUSSION THREAD

**Discussion by calllib** [OpenCCU/OpenCCU#1639] [🇩🇪 Fragen&Antworten] [answered]
Bei dem Versuch, mir in der **WebUI** die **Zusatzsoftware** anzeigen zu lassen (Startseite > Einstellungen > Systemsteuerung > Zusatzsoftware), bekomme ich immer eine auch bei gültigem Username/Password nicht funktionierende Login-Abfrage. **Die Anzeige dieser Seite hat früher ohne zusätzlichen Login funktioniert.**

Im lighttpd-error.log gibt es dann den Eintrag:
`2022-01-02 12:15:55: (mod_auth.c.840) password doesn't match for /addons/cuxd/CUXD.PNG username: ... IP: ...`
(... durch mich hier eingesetzt.) Bei CUXD.PNG handelt es sich um ein Icon, das un der rechten Spalte unterhalb des Textes "CUx-Daemon 2.7" angezeigt würde. Rechte im Dateisystem: 644

Was läuft da schief? Welche Konfiguration stimmt nicht?

![Screenshot 11](https://user-images.githubusercontent.com/16517367/147939269-65d4b47b-f5de-4e33-91ce-83fa26cea894.png)
![Screenshot2](https://user-images.githubusercontent.com/16517367/147939272-b25d90a4-6109-4d9e-9a9a-afcb69781644.png)

Dies ist in Teilen eine Wiederholung von [#1634](https://github.com/jens-maus/RaspberryMatic/discussions/1634), hier aber nach neueren Erkenntnissen allgemeiner gefasst, da es sich offenbar nicht um ein CUxD-Problem handelt, sondern vielleicht eher um eines der Webserver-Konfiguration.

(RaspberryMatic 3.61.7.20211218)

**Comment by jens-maus** [comment]
Welche Authentifizierungseinstellungen sind in der WebUI unter Sicherheit konfiguriert? Und bitte das mal auf einem Testsystem mit frischem RM versuchen nachzuspielen bzw reproduzieren.

**Reply by calllib** [reply]
Authentifizierung ist aktiv. Ich hatte es erfolglos auch schon mit deaktivierter probiert (aber ohne Neustart).
Ein Testsystem bekomme ich nicht so einfach aufgesetzt. Ich nehme auch an, dass es funktionieren würde, denn es handelt sich doch wahrscheinlich nicht um ein allgemeines Problem dieser Version mit dem CUxD, sonst gäbe es darüber bestimmt mehr Berichte hier.

In `/usr/local/etc/config/lighttpd/auth.conf` steht:
```auth.backend = "htpasswd" 
auth.backend.htpasswd.userfile = "/etc/config/lighttpd/.lighttpdpassword"
auth.require = (
...
        "/addons/cuxd" =>
        (
                "method" => "basic",
                "realm" => "api",
                "require" => "valid-user"
        )
)
```
Ist das korrekt?

Was mich frustriert ist, dass ich mit dem Troubleshooting nicht weiterkomme, weil ich weder Web-, noch sonstiger Programmierer bin. Ich bekomme nicht einmal das Debugging des lighttpd angeworfen. Die Datei `/etc/lighttpd/conf.d/debug.conf` ist auch für den User root offenbar in einem "Readonly"-Filesystem, und da hört mein Latein auf. (Ich weiß aber auch nicht, ob ich in Debug-Logs etwas für mich Verständliches entdecken würde...)

**Reply by jens-maus** [reply]
> In /usr/local/etc/config/lighttpd/auth.conf steht:
> 

Wo kommt die Datei denn bitte her? soweit ich weiss ist die nicht teil einer standard CUxD Installation. Hast du die selbst gebastelt?!?

**Reply by calllib** [reply]
Nein, die ist bestimmt nicht von mir. Hier ist die ganze Datei. Da da auch ein anderes Addon aufgeführt ist, ging ich davon aus, dass sie Ergebnis der RaspberryMatic-Installationsroutinen sei:
```
auth.backend = "htpasswd" 
auth.backend.htpasswd.userfile = "/etc/config/lighttpd/.lighttpdpassword"
auth.require = (
        "/config/xmlapi" =>
        (
                "method" => "basic",
                "realm" => "api",
                "require" => "valid-user"
        ),
        "/addons/xmlapi" =>
        (
                "method" => "basic",
                "realm" => "api",
                "require" => "valid-user"
        ),
		

		"/config/email" =>
        (
                "method" => "basic",
                "realm" => "api",
                "require" => "valid-user"
        ),
        "/addons/email" =>
        (
                "method" => "basic",
                "realm" => "api",
                "require" => "valid-user"
        ),
		
		
		"/config/cuxd" =>
        (
                "method" => "basic",
                "realm" => "api",
                "require" => "valid-user"
        ),
        "/addons/cuxd" =>
        (
                "method" => "basic",
                "realm" => "api",
                "require" => "valid-user"
        )
)
```

**Comment by Baxxy13** [comment, marked as answer]
Ich habe auf keinem meiner Testsysteme diese Dateien bzw. das Verzeichnis `/usr/local/etc/config/lighttpd/`
Mir ist auch kann AddOn bekannt was diese "Authentifikation" einrichtet.
Sicher das du das nicht selber warst? So Mitte 2018 vielleicht? 
Zumindest war / ist das "Template" in #332 genau so zu sehen.

**Reply by jens-maus** [reply]
Ich denke auch das sich @calllib hier irgendwie/irgendwann selbst ein Ei gelegt hat durch das anlegen dieser lighttpd config files und das fällt ihm eben jetzt auf die Füße! Ergo, einfach mal dieses ganze `/usr/local/etc/config/lighttpd` Verzeichnis löschen, Zentrale neustarten und schon wird es funktionieren.

**Reply by calllib** [reply]
Korrekt! Während ich mir die Datei nie hätte selbst ausdenken können, habe ich offenbar vor Jahren https://github.com/fl0wde/RaspberryMatic_Authentification ausprobiert, nicht rückgängig gemacht und den verursachten Fehler damals auch nicht bemerkt – vielleicht führte das damals auch gar nicht zu einem Fehler.
Wie Jens Maus schrieb, habe ich das Verzeichnis `/usr/local/etc/config/lighttpd` gelöscht, die Schachtel durchgestartet, und der Fehler ist weg.
Ganz großen Dank an @Baxxy13! Ihr seid beide super!
